homeassistant/components/nest/binary_sensor_sdm.py

```diff
@@ -0,0 +1,198 @@
+"""Support for Google Nest SDM binary_sensors."""
+
+import datetime
+import logging
+from typing import Optional
+
+from google_nest_sdm.camera_traits import (
+    CameraMotionTrait,
+    CameraPersonTrait,
+    CameraSoundTrait,
+)
+from google_nest_sdm.device import Device
+from google_nest_sdm.doorbell_traits import DoorbellChimeTrait
+from google_nest_sdm.event import (
+    CameraMotionEvent,
```

Comment:
We should not show motion events as binary sensors. Instead, they should be fired as events of type `nest_event` and the pubsub -> HA events logic should be instantiated from `async_setup_entry` in `__init__.py`, it should not be tied to an entity. 

In a future PR device triggers can be offered to allow easy automating it. Example of a ZHA remote (also event based):

![image](https://user-images.githubusercontent.com/1444314/98167566-902ccb00-1ee9-11eb-8a97-39002d7a668a.png)

Reply:
Thank you for the guidance and pointer for an example to follow.  I'm converting this to a draft while I update.